ephypype/compute_inv_problem.py

```diff
@@ -31,68 +33,91 @@ def _compute_noise_cov(cov_fname, raw):
-            the raw data
-
+            noise covariance file name template
+        raw_filename : str
+            raw filename
     Output
         cov_fname : str
             noise covariance file name in which is saved the noise covariance
             matrix
     """
-
-    print(('***** COMPUTE RAW COV *****' + cov_fname))
+    # Check if cov matrix exists
+    cov_fname = _get_cov_fname(fname_template)
 
     if not op.isfile(cov_fname):
+        er_raw, cov_fname = _get_er_data(fname_template)
 
-        data_path, basename, ext = split_f(raw.info['filename'])
-        fname = op.join(data_path, '%s-cov.fif' % basename)
-
-        reject = _create_reject_dict(raw.info)
+        if not op.isfile(cov_fname) and er_raw:
+            reject = _create_reject_dict(er_raw.info)
+            picks = pick_types(er_raw.info, meg=True,
+                               ref_meg=False, exclude='bads')
 
-        picks = pick_types(raw.info, meg=True, ref_meg=False, exclude='bads')
+            noise_cov = compute_raw_covariance(er_raw, picks=picks,
+                                               reject=reject)
 
-        noise_cov = compute_raw_covariance(raw, picks=picks, reject=reject)
-
-        write_cov(fname, noise_cov)
+            write_cov(cov_fname, noise_cov)
+        elif op.isfile(cov_fname):
+            print(('*** NOISE cov file {} exists!!!'.format(cov_fname)))
+        elif not er_raw:
+            cov_fname = compute_cov_identity(raw_filename)
 
     else:
-        print(('*** NOISE cov file %s exists!!!' % cov_fname))
+        print(('*** NOISE cov file {} exists!!!'.format(cov_fname)))
 
     return cov_fname
 
 
-def _read_noise_cov(cov_fname, raw_info):
-    """
-    Read a noise covariance matrix from cov_fname
+def _get_cov_fname(cov_fname_template):
+    "Check if a cov matrix already exists. If it exists return the cov \
+    matrix file name otherwhise returns."
+    for cov_fname in glob.glob(cov_fname_template):
+        if cov_fname.rfind('cov.fif') > -1:
+            return cov_fname
 
-    Inputs
-        cov_fname : str
-            noise covariance file name
-        raw_info : dict
-            dictionary containing the information about the raw data
+    return ''
 
-    Outputs
-        noise_cov : Covariance
-            the noise covariance matrix
-    """
 
-    print(('***** READ RAW COV *****' + cov_fname))
+def _get_er_data(er_fname_template):
+    "Check if empty room data exists in order to compute noise cov matrix. If \
+    it exists returns both the raw instance and cov filename where we'll save \
+    the cov matrix."
+    for er_fname in glob.glob(er_fname_template):
+        print(('*** {} \n'.format(er_fname)))
+
+        if er_fname.rfind('.fif') > -1:
+            er_raw = read_raw_fif(er_fname)
+            cov_fname = er_fname.replace('.fif', '-raw-cov.fif')
+        elif er_fname.rfind('.ds') > -1:
+            cov_fname = er_fname.replace('.ds', '-raw-cov.fif')
+            try:
```

Comment:
yikes! no `try` please :)

[do or do not, there is no try](https://www.youtube.com/watch?v=BQ4yd2W50No) ;-)

Reply:
> yikes! no `try` please :)
> 
> [do or do not, there is no try](https://www.youtube.com/watch?v=BQ4yd2W50No) ;-)

some weeks ago my friend sent me the same video in Italian :D